src/Asset/ContaoContext.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Contao\CoreBundle\Asset;
+
+use Symfony\Component\Asset\Context\ContextInterface;
+
+class ContaoContext implements ContextInterface
+{
+    /**
+     * @var string
+     */
+    private $constant;
+
+    /**
+     * Constructor.
+     *
+     * @param string $constant
+     */
+    public function __construct(string $constant)
+    {
+        $this->constant = $constant;
+    }
+
+    /**
+     * Gets the base path.
+     *
+     * @return string The base path
+     */
+    public function getBasePath()
+    {
+        if (!defined($this->constant)) {
+            return '';
+        }
```

Comment:
is this check needed? `constant` cannot be undefined due to constructor string requirement and no method which unsets the property